contracts/secret-storage/SecretManager.sol

```diff
@@ -0,0 +1,601 @@
+            if(userStorage[_secretId].selectedEnclaves[index].enclaveAddress == _enclaveAddress)
+                return index;
+        }
+        revert SecretManagerEnclaveNotFound();
+    }
+
+    function _acknowledgeStoreFailed(
+        uint256 _secretId
+    ) internal {
+        UserStorage memory userStoreData = userStorage[_secretId];
+        bool ackFailed;
+        uint256 len = userStoreData.selectedEnclaves.length;
+        for (uint256 index = 0; index < len; index++) {
+            if(block.timestamp <= userStoreData.selectedEnclaves[index].selectTimestamp + ACKNOWLEDGEMENT_TIMEOUT)
+                revert SecretManagerAcknowledgementTimeoutPending(userStoreData.selectedEnclaves[index].enclaveAddress);
+
+            if(!userStoreData.selectedEnclaves[index].hasAcknowledgedStore)
+                ackFailed = true;
+
+            SECRET_STORE.releaseEnclave(userStoreData.selectedEnclaves[index].enclaveAddress, userStoreData.sizeLimit);
+        }
+
+        if(!ackFailed)
+            revert SecretManagerAcknowledgedAlready();
+
+        delete userStorage[_secretId];
+        USDC_TOKEN.safeTransfer(userStoreData.owner, userStoreData.usdcDeposit);
+
+        emit SecretStoreAcknowledgementFailed(_secretId);
+    }
+
+    function _markStoreAlive(
+        uint256 _signTimestamp,
+        bytes memory _signature
+    ) internal {
+        address enclaveAddress = _verifyStoreAliveSign(_signTimestamp, _signature);
+
+        ( ,uint256 storageOccupied, , uint256 lastAliveTimestamp, address owner, ) = SECRET_STORE.secretStorage(enclaveAddress);
+        SECRET_STORE.updateLastAliveTimestamp(enclaveAddress, _signTimestamp);
+
+        uint256 usdcPayment = (block.timestamp - lastAliveTimestamp) * storageOccupied * SECRET_STORE_FEE_RATE;
+        USDC_TOKEN.safeTransfer(owner, usdcPayment);
+
+        emit SecretStoreAlive(enclaveAddress);
+    }
+
+    function _verifyStoreAliveSign(
+        uint256 _signTimestamp,
+        bytes memory _signature
+    ) internal view returns(address signer) {
+        _checkSignValidity(_signTimestamp);
+
+        bytes32 hashStruct = keccak256(abi.encode(ALIVE_TYPEHASH, _signTimestamp));
+        bytes32 digest = keccak256(abi.encodePacked("\x19\x01", DOMAIN_SEPARATOR, hashStruct));
+        signer = digest.recover(_signature);
+
+        SECRET_STORE.allowOnlyVerified(signer);
+    }
+
+    function _markStoreDead(
+        address _enclaveAddress,
+        uint256[] memory _secretIds
+    ) internal {
+        // FIX: get store data
+        ( , , , uint256 lastAliveTimestamp, , ) = SECRET_STORE.secretStorage(_enclaveAddress);
+        if(block.timestamp <= lastAliveTimestamp + MARK_ALIVE_TIMEOUT)
+            revert SecretManagerStoreIsAlive();
+
+        for (uint256 index = 0; index < _secretIds.length; index++) {
+            uint256 enclaveIndex = _getSelectedEnclaveIndex(_secretIds[index], _enclaveAddress);
+            _markEnclaveDead(_secretIds[index], _enclaveAddress, enclaveIndex);
```

Comment:
remove the encalveIndex arg from _markEncalveDead and move the _getSelectedEnclaveIndex inside it.